knotx-knot/knotx-knot-action/src/main/java/io/knotx/knot/action/FormEntity.java

```diff
@@ -0,0 +1,137 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.knotx.knot.action;
+
+import static io.knotx.knot.action.FormConstants.FORM_ACTION_ATTR;
+import static io.knotx.knot.action.FormConstants.FORM_DEFAULT_IDENTIFIER;
+import static io.knotx.knot.action.FormConstants.FORM_SIGNAL_ATTR_PREFIX;
+import static io.knotx.knot.action.FormConstants.FRAGMENT_KNOT_PATTERN;
+import static io.knotx.knot.action.FormConstants.FRAGMENT_KNOT_PREFIX;
+
+import io.knotx.dataobjects.Fragment;
+import io.knotx.exceptions.ConfigurationException;
+import io.knotx.fragments.FragmentContentExtractor;
+import io.knotx.knot.action.ActionKnotConfiguration.AdapterMetadata;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.StringUtils;
+import org.jsoup.nodes.Document;
+
+public class FormEntity {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(FormEntity.class);
+
+  private Fragment fragment;
+
+  private String identifier;
+
+  private AdapterMetadata adapter;
+
+  private Map<String, String> signalToUrl;
+
+  public static FormEntity from(Fragment fragment, ActionKnotConfiguration configuration) {
+    Document scriptDocument = FragmentContentExtractor.getUnwrappedDocument(fragment);
+    return new FormEntity()
+        .fragment(fragment)
+        .identifier(getFormIdentifier(fragment))
+        .adapter(getAdapterMetadata(configuration, getAdapterName(fragment, scriptDocument)))
+        .signalToUrlMapping(getSignalToUrlMapping(scriptDocument));
+
+  }
+
+  public Fragment getFragment() {
+    return fragment;
+  }
+
+  public String getIdentifier() {
+    return identifier;
+  }
+
+  public AdapterMetadata getAdapter() {
+    return adapter;
+  }
+
+  public Optional<String> getUrl(String signal) {
+    return Optional.ofNullable(signalToUrl.get(signal));
+  }
+
+  private FormEntity fragment(Fragment fragment) {
+    this.fragment = fragment;
+    return this;
+  }
+
+  private FormEntity identifier(String formIdentifier) {
+    this.identifier = formIdentifier;
+    return this;
+  }
+
+  private FormEntity adapter(AdapterMetadata adapterMetadata) {
+    this.adapter = adapterMetadata;
+    return this;
+  }
+
+  private FormEntity signalToUrlMapping(Map<String, String> signalToUrlMapping) {
+    this.signalToUrl = signalToUrlMapping;
+    return this;
+  }
+
+  private static String getFormIdentifier(Fragment fragment) {
```

Comment:
How about providing some simple unit test for this and methods below?
It would require making them package scope or protected though.